### ISSUE TITLE
`__torch_function__` docs should use @classmethod

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#63767]
## 📚 Documentation

https://pytorch.org/docs/master/notes/extending.html?highlight=extending

@ezyang told me that non-classmethod uses of `__torch_function__` should be considered legacy. Some of the examples still use the method `__torch_function__` and should be updated to encourage new users to use the classmethod variant.

**Comment by albanD** [comment]
How should code like the following be updated if these are not plain methods anymore? https://github.com/pytorch/pytorch/blob/5060b69d62943e11f053f8c86b8154a7d009af11/test/test_overrides.py#L717-L722

Should this just pick a Tensor in the args at random?

**Comment by hameerabbasi** [comment]
> How should code like the following be updated if these are not plain methods anymore? https://github.com/pytorch/pytorch/blob/5060b69d62943e11f053f8c86b8154a7d009af11/test/test_overrides.py#L717-L722
> 
> 
> 
> Should this just pick a Tensor in the args at random?
> 
> 
> 
> cc @ezyang because we were talking about this offline.

Since Python is dynamic, you can just shove everything into the type, and create instances via a metaclass. But I'm being extreme here.

**Comment by albanD** [comment]
A simple solution would be to scan the args for the first subclass of this class.
My question is more: will forcing this to be a subclass actually be a problem for some use case?

**Comment by hameerabbasi** [comment]
No, that isn't the case here. `self` is guaranteed to be in `args` somewhere.

**Comment by albanD** [comment]
meta question for the PR above: should we make `pytree` a public API now? And advertise it as "the right way" to extract Tensor subclasses from the args/kwargs?

**Comment by ezyang** [comment]
> meta question for the PR above: should we make pytree a public API now? And advertise it as "the right way" to extract Tensor subclasses from the args/kwargs?

Yes probably.

**Comment by zou3519** [comment]
The reason why it isn't public is because we never tested it for performance, but I agree we should make it public to support this use case
